code/__HELPERS/antag_check.dm

```diff
@@ -0,0 +1,60 @@
+#define ANTAGONISM_MINIMAL_TIME 180 // 1 is identical to 1 min in DB
+
+/// checks if antagonism configuration has been set
+/proc/check_antagonism_config(mob/M)
+	if (!CONFIG_GET(flag/use_exp_tracking))
+		return FALSE // if we don't exp tracking, no way to track playtime. returns FALSE
+	if (!CONFIG_GET(flag/use_exp_track_for_antagonism_behavior))
+		return FALSE // if this system is not used in config, returns FALSE
+	if(length(M.mind.antag_datums))
+		return FALSE // antags shouldn't be restricted by this.
+	return TRUE
```

Comment:
Ugh, this could get annoying for situations where someone is allowed to do something by some edge case scenario but then this still returns true. Theres just a lot of possibility for errors by going with this method, making me think there is a much better solution out there that is a bit better designed and integrates better with the game (But I do like the concept of mechanically stopping people from purposely griefing (But I also think that people should still be able to make mistakes and screw up)).

Reply:
then I will make the system can be used more generally.